Shift Left: Try building and testing selective_build in OSS

mergennachin opened the pull request:
- Enable buck targets in OSS under codegen/tools/targets.bzl
- Build all buck targets in OSS under codegen/tools/... (Can't test yet due to torch not available in OSS buck)
- Run test_tools_selective_build.py in pytest
- Building on top of https://github.com/pytorch/executorch/pull/14488

lucylq reviewed (approved):
Thanks for these changes 🙏

mergennachin committed:
Shift Left: Try building and testing selective_build in OSS (#14499)

Summary:
- Enable buck targets in OSS under codegen/tools/targets.bzl
- Build all buck targets in OSS under codegen/tools/... (Can't test yet due to torch not available in OSS buck)
- Run test_tools_selective_build.py in pytest
- Building on top of https://github.com/pytorch/executorch/pull/14488

Pull Request resolved: https://github.com/pytorch/executorch/pull/14499

Reviewed By: lucylq

Differential Revision: D83024265

Pulled By: mergennachin
